lib/diagnostic.js

```diff
@@ -5,17 +5,28 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function (previous, current) {
+  const workingArray = previous.slice();
```

Comment:
```previous``` and ```current``` are assumed to be numbers so here you're trying to invoke ```slice()``` on a number, which will throw an error.